linux/check_os_fullname/validate_os_fullname.yml

```diff
@@ -45,8 +40,11 @@
         - name: "Test passed on ESXi {{ esxi_version }}"
           ansible.builtin.meta: end_host
       when:
-        - unmapped_guest_fullname is defined and unmapped_guest_fullname
-        - unmapped_guest_fullname in guestinfo_guest_full_name
+        - guest_os_ansible_kernel and guest_os_ansible_kernel in guestinfo_guest_full_name
+        - ((guest_os_release.PRETTY_NAME is defined and guest_os_release.PRETTY_NAME and
+            guest_os_release.PRETTY_NAME in guestinfo_guest_full_name) or
+           (guest_os_release.NAME is defined and guest_os_release.NAME and
+            guest_os_release.NAME in guestinfo_guest_full_name))
   when:
```

Comment:
Move when condition to line 25. If the outer block is no needed, please merge it with the inner block condition.